source/common/http/async_client_impl.cc

```diff
@@ -29,92 +28,92 @@ AsyncClientImpl::~AsyncClientImpl() {
 
 AsyncClient::Request* AsyncClientImpl::send(MessagePtr&& request, AsyncClient::Callbacks& callbacks,
                                             const Optional<std::chrono::milliseconds>& timeout) {
-  std::unique_ptr<AsyncRequestImpl> new_request{
-      new AsyncRequestImpl(std::move(request), *this, callbacks, timeout)};
+  AsyncRequestImpl* async_request =
+      new AsyncRequestImpl(std::move(request), *this, callbacks, timeout);
+  std::unique_ptr<AsyncStreamImpl> new_request{async_request};
 
   // The request may get immediately failed. If so, we will return nullptr.
   if (!new_request->complete_) {
     new_request->moveIntoList(std::move(new_request), active_requests_);
-    return active_requests_.front().get();
+    return async_request;
   } else {
     return nullptr;
   }
 }
 
-AsyncRequestImpl::AsyncRequestImpl(MessagePtr&& request, AsyncClientImpl& parent,
-                                   AsyncClient::Callbacks& callbacks,
-                                   const Optional<std::chrono::milliseconds>& timeout)
-    : request_(std::move(request)), parent_(parent), callbacks_(callbacks),
-      stream_id_(parent.config_.random_.random()), router_(parent.config_),
-      request_info_(Protocol::Http11), route_(parent_.cluster_.name(), timeout) {
+AsyncClient::Stream* AsyncClientImpl::start(AsyncClient::StreamCallbacks& callbacks,
+                                            const Optional<std::chrono::milliseconds>& timeout) {
+  AsyncStreamImpl* async_request = new AsyncStreamImpl(*this, callbacks, timeout);
+  std::unique_ptr<AsyncStreamImpl> new_request{async_request};
```

Comment:
nit: combine into one line
std::unique_ptr<AsyncStreamImpl> new_request(new AsyncStreamImpl(*this, callbacks, timeout));